clang/docs/SYCLFERFCIntegrationHeaderFooter.md

```diff
@@ -0,0 +1,179 @@
+# Integration Header
+
+We propose an Integration Header for two purposes:
+
+* to allow a non-SYCL-aware compiler to do host compilations and
+* as a mechanism permitting the SYCL run-time library to use the information provided for its needs
+
+Consider the following partial code:
+
+```
+  sycl::accessor<char, 1, sycl::access::mode::read> acc1;
+  int i = 13;
+  sycl::sampler smplr;
+  struct S {
+    char c;
+    int i;
+  } test_s;
+  test_s.c = 14;
+  kernel_single_task<class first_kernel>([=]() {
+    if (i == 13 && test_s.c == 14) {
+
+      acc1.use();
+      smplr.use();
+    }
+  });
+```
+
+Here the lambda within the 'kernel_single_task' construct needs to be executed on the device.  To do this, a function object representing the kernel is generated.   The function call operator of this object has the contents of the kernel invocation.  The function object generated looks like:
```

Comment:
>> the lambda within the 'kernel_single_task' construct

lambda passed  to*

>>  To do this, a function object representing the kernel is generated.

I don't think we need to say this. We don't create this function object specially. It is how lambdas are represented by the compiler. I think we can just say something like the function object corresponding to this kernel lambda is as follows and give the example below. We can then move this line "The function call operator of this object has the contents of the kernel invocation." to after the example

Reply:
I agree. More generally, I believe the lambda object is constructed by the host (because captures have to be processed there) and then passed as an argument to the synthesized kernel function which then invokes its member function call operator. My understanding is that the integration header facilitates the call to the synthesized kernel function on the (correct) device.

Reply:
Thank you both, done.